File: joiner-test-support/src/main/java/cz/encircled/joiner/test/model/Address.java
package cz.encircled.joiner.test.model;

import javax.persistence.*;
import java.util.Set;

/**
 * @author Kisel on 25.01.2016.
 */
@Entity
@Table(name = "test_address")
public class Address extends AbstractEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", unique = false)
    private User user;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "address")
    private Set<Status> statuses;

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Set<Status> getStatuses() {
        return statuses;
    }

    public void setStatuses(Set<Status> statuses) {
        this.statuses = statuses;
    }
}

File: joiner-test-support/src/main/java/cz/encircled/joiner/test/model/Address.kt
package cz.encircled.joiner.test.model

import javax.persistence.*

/**
 * @author Kisel on 25.01.2016.
 */
@Entity
@Table(name = "test_address")
class Address : AbstractEntity() {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var user: User? = null

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "address")
    var statuses: Set<Status>? = null
}
